fix: add timeout to prevent delete_session hangs

shanchunyang0919 opened the pull request:
# Description

In group conversations, it appears that as the number of recipients increases, the SLIM delete_session method can occasionally hang indefinitely, even though the session is successfully deleted. This hang can cause the services to become unresponsive. To prevent the app-sdk from accidentally hanging while closing the session, we have added a timeout to ensure proper handling.

## Type of Change

- [X] Bugfix
- [ ] New Feature
- [ ] Breaking Change
- [ ] Refactor
- [ ] Documentation
- [ ] Other (please describe)

## Checklist

- [ ] I have read the [contributing guidelines](/agntcy/app-sdk/blob/main/CONTRIBUTING.md)
- [ ] Existing issues have been referenced (where applicable)
- [ ] I have verified this change is not present in other open pull requests
- [ ] Functionality is documented
- [ ] All code style checks pass
- [ ] New code contribution is covered by automated tests
- [X] All new and existing tests pass

Shridhar-2205 reviewed (approved):
LGTM, just bump the sdk

panyuenlau commented on uv.lock:
Why are there so many diffs in this file? It seems like you ran `uv sync` against your local dependencies?

shanchunyang0919 replied:
Thanks for bringing this up.

The uv.lock file appears to be outdated. I ran end‑to‑end tests with Lungo (including the Weather MCP server, since the Anthropic library shows a major version bump in uv.lock), and everything looks fine.

shanchunyang0919 commented:
> Is this one of the trial & error attempts or the recommendation from SLIM team?

Yes, but I reverted that change because their fix does not resolve the issue and also causes a hang in the simple `create_client` call. 

The small timeout fix I added earlier is sufficient for the help-desk agent to function. 

I'll add a note in the doc that the new help-desk agent is not designed to handle concurrent requests. We can wait for SLIM to address the session closing issue in 0.7.0.

shanchunyang0919 commented:
> > > Is this one of the trial & error attempts or the recommendation from SLIM team?
> > 
> > 
> > Yes, but I reverted that change because their fix does not resolve the issue and also causes a hang in the simple `create_client` call.
> > The small timeout fix I added earlier is sufficient for the help-desk agent to function.
> > I'll add a note in the doc that the new help-desk agent is not designed to handle concurrent requests. We can wait for SLIM to address the session closing issue in 0.7.0.
> 
> @shanchunyang0919 - Do we have to make this session delete operation a blocking operation? If we make it a non-blocking operation - just ask the SLIM node to delete this session and move on, will this resolve the issue?

Both the app-sdk `close_session` async-call and the SLIM call `self._slim.delete_session(session.id)` are non-blocking in structure. The likely problem is inside the SLIM binding’s `delete_session` implementation: it may wait for an deletion event that sometimes never arrives. In those cases the remote session is actually deleted, but the Python await never completes.
 
Here is what Mauro says about the `delete_session` issue: the hang happened because the Python coroutine was waiting on a Rust future that never resolved (see more info here: https://github.com/PyO3/pyo3-async-runtimes), so there was no remaining stack progress and the await never finished.

Regarding your second point: my understanding is that self._slim.delete_session(session.id) already instructs the SLIM node to delete the session. Please let me know if I have misunderstood.

panyuenlau commented:
> > > > Is this one of the trial & error attempts or the recommendation from SLIM team?
> > > 
> > > 
> > > Yes, but I reverted that change because their fix does not resolve the issue and also causes a hang in the simple `create_client` call.
> > > The small timeout fix I added earlier is sufficient for the help-desk agent to function.
> > > I'll add a note in the doc that the new help-desk agent is not designed to handle concurrent requests. We can wait for SLIM to address the session closing issue in 0.7.0.
> > 
> > 
> > @shanchunyang0919 - Do we have to make this session delete operation a blocking operation? If we make it a non-blocking operation - just ask the SLIM node to delete this session and move on, will this resolve the issue?
> 
> Both the app-sdk `close_session` async-call and the SLIM call `self._slim.delete_session(session.id)` are non-blocking in structure. The likely problem is inside the SLIM binding’s `delete_session` implementation: it may wait for an deletion event that sometimes never arrives. In those cases the remote session is actually deleted, but the Python await never completes.
> 
> Here is what Mauro says about the `delete_session` issue: the hang happened because the Python coroutine was waiting on a Rust future that never resolved (see more info here: https://github.com/PyO3/pyo3-async-runtimes), so there was no remaining stack progress and the await never finished.
> 
> Regarding your second point: my understanding is that self._slim.delete_session(session.id) already instructs the SLIM node to delete the session. Please let me know if I have misunderstood.

thanks for the info!

panyuenlau commented on pyproject.toml:
```diff
@@ -18,7 +18,7 @@ dependencies = [
     "opentelemetry-instrumentation-starlette>=0.54b0",
     "asgi-lifespan>=2.1.0",
-    "slim-bindings==0.4.1",
+    "slim-bindings==0.5.0",
```
Do we have to use 0.5.0? If not, can we keep it at v0.4.x since the support for 0.6.0 is already on the way
